packages/framework/src/Model/Product/ProductDetailFriendlyUrlDataProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Model\Product;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\CompilerPass\FriendlyUrlDataProviderInterface;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrl;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlData;
+
+class ProductDetailFriendlyUrlDataProvider implements FriendlyUrlDataProviderInterface
+{
+    const ROUTE_NAME = 'front_product_detail';
+
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    private $em;
+
+    public function __construct(EntityManagerInterface $em)
```

Comment:
I am wondering, how it happens that in other implementations you use different formatting that here? I do not mind it much but I am curious :laughing: Personally I like the inline styling better than this (but the styling below is better for future diffs...):
```
public function __construct(
        EntityManagerInterface $em
    ) {
        $this->em = $em;
    }
```